org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/codemanipulation/StubUtility.java

```diff
@@ -204,6 +207,47 @@ public static String getCatchBodyContent(ICompilationUnit cu, String exceptionTy
 		context.setVariable(CodeTemplateContextType.EXCEPTION_VAR, variableName);
 		return evaluateTemplate(context, template);
 	}
+
+	public static String getClassSnippetContent(ICompilationUnit cu, String lineDelimiter) throws CoreException {
+		CodeGenerationTemplate templateSetting = CodeGenerationTemplate.CLASSSNIPPET;
+		Template template = templateSetting.createTemplate(cu.getJavaProject());
+		if (template == null) {
+			return null;
+		}
+		CodeTemplateContext context = new CodeTemplateContext(template.getContextTypeId(), cu.getJavaProject(), lineDelimiter);
+		setVariables(context, cu);
+		return evaluateTemplate(context, template);
+	}
+
+	public static String getInterfaceSnippetContent(ICompilationUnit cu, String lineDelimiter) throws CoreException {
+		CodeGenerationTemplate templateSetting = CodeGenerationTemplate.INTERFACESNIPPET;
+		Template template = templateSetting.createTemplate(cu.getJavaProject());
+		if (template == null) {
+			return null;
+		}
+		CodeTemplateContext context = new CodeTemplateContext(template.getContextTypeId(), cu.getJavaProject(), lineDelimiter);
+		setVariables(context, cu);
+		return evaluateTemplate(context, template);
+	}
+
+	private static void setVariables(CodeTemplateContext context, ICompilationUnit cu) throws CoreException {
+		IPackageDeclaration[] packageDeclartions = cu.getPackageDeclarations();
+		String packageHeader = (packageDeclartions == null || packageDeclartions.length == 0) ? "package " + cu.getParent().getElementName() + ";\n\n" : "";
+		context.setVariable(CodeTemplateContextType.PACKAGEHEADER, packageHeader);
+		String typeName = JavaCore.removeJavaLikeExtension(cu.getElementName());
+		List<IType> types = Arrays.asList(cu.getTypes());
+		int postfix = 0;
+		while (types != null && !types.isEmpty() && types.stream().filter(isTypeExists(typeName)).findFirst().isPresent()) {
+			typeName = "Inner" + JavaCore.removeJavaLikeExtension(cu.getElementName()) + (postfix == 0 ? "" : "_" + postfix);
```

Comment:
**IF** code snippet is supported by the client (see ClientPreferences.isCompletionSnippetsSupported), then we should return `"${1:Inner"+typeName+"}", so users can change the class name instantly